ext/curl/interface.c

```diff
@@ -1547,6 +1547,8 @@ static size_t curl_read(char *data, size_t size, size_t nmemb, void *ctx)
 				if (Z_TYPE(retval) == IS_STRING) {
 					length = MIN((int) (size * nmemb), Z_STRLEN(retval));
 					memcpy(data, Z_STRVAL(retval), length);
+				} else if (Z_TYPE(retval) == IS_LONG) {
```

Comment:
would it be difficult to add a test for the usage of `CURL_READFUNC_PAUSE`?

Reply:
I added one, that I know test the code, but what I don't like about it is that there is no way to know that the read is paused. But I guess it's better than nothing.

Reply:
Yeah, it's definitely better! Would it make sense to return `CURL_READFUNC_PAUSE` until a specific amount of time elapses (e.g. 1 sec)? If this makes sense then the test should be tagged as slow

Reply:
The `CURLOPT_READFUNCTION` callback is not called after returning `CURL_READFUNC_PAUSE` until `curl_pause(CURLPAUSE_CONT)` is called. So I don't think that it will really improve the test.

Reply:
OK then!